Feature admin

WebDevJL opened the pull request:
@damirius There is a lot of untested code here. Can you write the tests for it? Thanks.

damirius committed:
#57
Added custom authenticator listener, user provider, user and api service with tests

damirius committed:
#57
Added parameters copy for travis and fixed exception throw if api url isn't set

damirius committed:
#71
Added /admin/customerinforequests page with overview of all customer info requests with filters and status update for them.

Commit:
Merge pull request #139 from Puzzlout/dev

dev to feature

damirius commented:
@WebDevJL 

I'll start writing new tests. Only problem I have is that when I'm testing controllers they are making calls to the API, I'll have to think of someway to get around that since we don't want to make real calls. Also travis will not work like that since it requires external api for it to work, so I have to create some mock data somehow and tell him to use that

damirius committed:
#138
Added unit tests for each Security class in AdminBundle

damirius committed:
#138
Added functional tests for AdminController customerinforequests list and status update and contact form, added some stuff to security unit tests

damirius commented:
@WebDevJL 

Hey, it should be 100% now: https://codecov.io/gh/Puzzlout/TrekToursFlyer/branch/feature_admin
I managed to mock everything we need, so it should be working great.

WebDevJL commented:
@damirius Great! I guess it can be merged then. However check out https://github.com/Puzzlout/TrekToursApi/issues/6 because I can't make it work locally...
